service-tests/travis.js

```diff
@@ -4,9 +4,43 @@ const Joi = require('joi');
 const ServiceTester = require('./runner/service-tester');
 const {isPhpVersionReduction} = require('./helpers/validators');
 
-const t = new ServiceTester({ id: 'travis', title: 'PHP version from .travis.yml' });
+const t = new ServiceTester({ id: 'travis', title: 'Travis CI/PHP version from .travis.yml' });
 module.exports = t;
 
+// Travis CI
+
+t.create('build status on default branch')
+  .get('/rust-lang/rust.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'build',
+    value: Joi.equal('failing', 'passing', 'unknown')
+  }));
+
+t.create('build status on named branch')
+  .get('/rust-lang/rust/stable.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'build',
+    value: Joi.equal('failing', 'passing', 'unknown')
+  }));
+
+t.create('unknown repo')
+  .get('/this-repo/does-not-exist.json')
+  .expectJSON({ name: 'build', value: 'unknown' });
+
+t.create('missing content-disposition header')
+  .get('/foo/bar.json')
+  .intercept(nock => nock('https://api.travis-ci.org')
+    .head('/foo/bar.svg')
+    .reply(200))
+  .expectJSON({ name: 'build', value: 'invalid' });
+
+t.create('connection error')
+  .get('/foo/bar.json')
+  .networkOff()
+  .expectJSON({ name: 'build', value: 'invalid' });
```

Comment:
usually we'd use 'inaccessible' for this case and 'invalid' when the response is malformed

Reply:
Looks good to merge when this is addressed!

Reply:
Thanks @chris48s, great idea. @paulmelnikow done